File: tspringmvc/src/main/java/com/tspringmvc/controllers/IndexController.java
package com.tspringmvc.controllers;

import com.tspringmvc.entity.User;
import com.tspringmvc.services.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpServletResponse;
import java.util.List;

@Controller
@RequestMapping(path = "/")
public class IndexController {
    private UserService userService;

    @Autowired
    public void setUserService(UserService userService) {
        this.userService = userService;
    }

    @GetMapping(path = "/all")
    public @ResponseBody
    List<User> findAllUsers() {
        return userService.findAll();
    }

    @PostMapping(path = "/add")
    public void saveUser(@RequestBody User user, HttpServletResponse res) {
        userService.save(user);
        res.setStatus(HttpServletResponse.SC_OK);
    }

    @PostMapping(path = "/delete/{id}")
    public void deleteUserById(@PathVariable("id") Long id, HttpServletResponse res) {
        userService.delete(id);
        res.setStatus(HttpServletResponse.SC_OK);
    }
}

File: tboot2kot/src/main/kotlin/com/tboot2kot/controllers/IndexController.kt
package com.tboot2kot.controllers

import com.tboot2kot.entity.User
import com.tboot2kot.services.IndexServiceImpl
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpHeaders
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = ["/"])
class IndexControllers(@Autowired private var indexServiceImpl: IndexServiceImpl,
                       @Autowired private var httpHeaders: HttpHeaders) {

    @GetMapping
    fun findAll(): ResponseEntity<List<User>> {
        httpHeaders.add("Access-Control-Allow-Origin", "*")
        return ResponseEntity(indexServiceImpl.findAllUsers(), httpHeaders, HttpStatus.CREATED)
    }
}